src/Tizen.NUI/src/public/BaseComponents/ViewAccessibility.cs

```diff
@@ -451,14 +457,21 @@ private IntPtr DuplicateString(string value)
             return Interop.ControlDevel.DaliToolkitDevelControlAccessibleImplNUIDuplicateString(value ?? "");
         }
 
-        private IntPtr DuplicateStates(AccessibilityStates states)
+        private IntPtr DuplicateRange(AccessibilityRange range)
         {
-            return Interop.ControlDevel.DaliToolkitDevelControlConvertState((ulong)states);
+            return Interop.ControlDevel.DaliAccessibilityNewRange(range.StartOffset, range.EndOffset, range.Content);
         }
 
-        private IntPtr DuplicateRange(AccessibilityRange range)
+        private static AccessibilityStates AccessibilityInitialStates = new AccessibilityStates();
+
+        private ulong AccessibilityCalculateStatesWrapper(ulong initialStates)
         {
-            return Interop.ControlDevel.DaliAccessibilityNewRange(range.StartOffset, range.EndOffset, range.Content);
+            lock (AccessibilityInitialStates)
```

Comment:
Why do we need 'lock' here?

Reply:
I made `AccessibilityInitialStates` `static` in order not to use up extra memory per every `View` instance. The AT-SPI bridge in dali-adaptor calls all `Accessible` methods sequentially from a single thread, so the code would work just as well without the `lock`. However, `AccessibilityInitialStates` still technically is a shared resource, and there is a possible race condition with someone else calling `GetAccessibilityStates` from a different thread.

Reply:
OK. Cool. There is a try to make separated UI thread as a this year project.